packages/iov-bcp-types/types/transactions.d.ts

```diff
@@ -61,4 +68,9 @@ export interface SwapTimeoutTx extends BaseTx {
     readonly kind: TransactionKind.SwapTimeout;
     readonly swapId: SwapIdBytes;
 }
-export declare type UnsignedTransaction = SendTx | SetNameTx | SwapOfferTx | SwapCounterTx | SwapClaimTx | SwapTimeoutTx;
+export interface RegisterUsernameTx extends BaseTx {
+    readonly kind: TransactionKind.RegisterUsername;
+    readonly username: string;
+    readonly addresses: Map<ChainId, Address>;
```

Comment:
I kept reading that tagged types didn't work as keys in a map.
So I was always using sting.
Nice if you can keep that safety here